### DISCUSSION TITLE
Recommended Hardware Specs to run rmf_demos

### DISCUSSION THREAD

**Discussion by albertleng** [osrf/rmf_demos#218] [Q&A]
I ran `rmf_demos`  via [Office-World](https://github.com/osrf/rmf_demos#office-world) in a VM in VMWare Workstation 15.5.  
Summary of my Virtual Machine:  

*OS*: **Ubuntu 20.04.2 LTS**
*Kernel Version*: **Linux 5.8.0-44-generic x86_64**   

*Memory*: **16GB**  
*Processor*: **8** *(Number of processors: 4, Number of cores per processor: 2)*   

The host machine's specs:  
*Processor*: **Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz 2.59 GHz**
*RAM*: **32.0 GB**  

I ran the `rmf_demos` few times but each time, the VM (virtual machine) hung. `RViz`, `Gazebo` etc. were started but the VM would hang after few minutes. Each time the VM got hung, I needed to force-shutdown and restarted again. I haven't really executed anything yet before it got hung. 

Attached are some logs for your reference. You can also view [20210309 RMF Demos Hang](https://youtu.be/n3O0OL40mE8) for more info.

Can you share with me the recommended specs (or the specs of any working machine) to run `rmf_demos`?

Thanks.
[launch_01.log](https://github.com/osrf/rmf_demos/files/6105856/launch_01.log)
[launch_02.log](https://github.com/osrf/rmf_demos/files/6105857/launch_02.log)

**Comment by Yadunund** [comment]
Hi @albertleng,

The specifications of your system are definitely sufficient. However, I suspect running on a VM is throttling performance. In general, we do not recommend running on a VM for this reason. Are you able to try the demos on a normal machine instead?

I also see another error in your launch. At the [10s](https://youtu.be/n3O0OL40mE8?t=10) mark, you can see that the dashboard crashes as you have not installed the `flask_cors` dependency. Could you run 
```python3 -m pip install Flask flask-socketio flask-cors```

I noticed that the installation [documentation](https://github.com/osrf/rmf_demos/blob/master/docs/installation.md#install-rmf-panel-dashbaord) is not updated so I apologise for that.

With the dependency installed, sending the `delivery` request as you did in the video should begin executing. However the slowdown you are experiencing will still remain as you are running on a VM.
